src/main/java/com/epam/processor/DataProcessor.java

```diff
@@ -93,31 +154,64 @@ public RoadAccident getAccidentByIndex(String index){
      * @return
      */
     public Collection<RoadAccident> getAccidentsByLocation(float minLongitude, float maxLongitude, float minLatitude, float maxLatitude){
-        return null;
+        return roadAccidentList.stream()
+        		.filter(r -> r.getLongitude() > minLongitude).filter(r -> r.getLongitude() < maxLongitude)
+        		.filter(r -> r.getLatitude() > minLatitude).filter(r -> r.getLatitude() < maxLatitude)
+        		.collect(Collectors.toList());
     }
 
     /**
      * find the weather conditions which caused max number of incidents
      * @return
      */
     public List<String> getTopThreeWeatherCondition(){
-        return null;
+		Map<String, Integer> map = new HashMap<String, Integer>();
+		roadAccidentList.stream().map(RoadAccident::getWeatherConditions).forEach(r -> {
+			Integer count = map.get(r);
+			if (count == null)
+				map.put(r, 1);
+			else
+				map.put(r, count + 1);
+		});
+
+		return map.entrySet().stream().sorted(new Comparator<Map.Entry<String, Integer>>() {
+			@Override
+			public int compare(Entry<String, Integer> o1, Entry<String, Integer> o2) {
+				return o2.getValue() - o1.getValue();
+			}
+		}).map(Entry::getKey).collect(Collectors.toList()).subList(0, 3);
```

Comment:
One more variant - use .limit(3)  before map
Better to cut collection as fast as you can, before any other operations.